File: potd/5-16-2024/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean evaluateTree(TreeNode root) {
        if (root.left == null && root.right == null) return root.val == 1;
        boolean leftVal = evaluateTree(root.left);
        boolean rightVal = evaluateTree(root.right);
        return (root.val == 2) ? leftVal || rightVal : leftVal && rightVal;
    }
}

File: potd/5-16-2024/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def evaluateTree(self, root: Optional[TreeNode]) -> bool:
        if (not root.left) and (not root.right): return root.val
        leftVal = self.evaluateTree(root.left)
        rightVal = self.evaluateTree(root.right)
        return leftVal or rightVal if root.val == 2 else leftVal and rightVal